### DISCUSSION TITLE
Store regExpr to fetch data from web-site into an own "service" to avoid the necessity of a release if changes happen

### DISCUSSION THREAD

**Discussion by murggel** [cgeo/cgeo#11240] [Ideas]
BTW - how about to move those regExpr into an own "service" to avoid the necessity of a release if those changes happen...

_Originally posted by @murggel in https://github.com/cgeo/cgeo/issues/11232#issuecomment-880154479_

Then only the service has to be updated but not all the user-installations

**Comment by murggel** [comment]
Or would it be possible to have a config-file and only this can be updated in such a case?

**Comment by fm-sys** [comment]
For sure, such config changes could theoretically be rolled out via our status message server if we implement it that way in c:geo. But I don't know if this could be a critical security issue in theory if someone would tries to hack us and introduce some malicious regex...?!

**Comment by baiti** [comment]
let me pick up the thought of "a service". The idea is to make the cgeo binary resilient against "scraping changes". So we could instantiate a server side proxy that implements a downstream API to cgeo and an upstream scraping engine.

Changes to the UI would only impact the proxy. No change to the installed cgeo binary would be necessary - ever.

In theory, we could design the downstream API semantically equivalent to what is known as GC HQ's "partner API", which would allow us to seamlessly switch to that technology if and once other surrounding circumstances have been resolved appropriately.

Such a proxy would have to meet some not so trivial requirements though. First of all, it is a single point of failure. If it is down, cgeo would be essentially unusable for all users. Second, it would have to meet significant scalability requirements as 10k to 50k users could be sending requests to that proxy in busy times in parallel. Last not least, each API opens up an attack vector for potential abuse. Plus there might be legal issues then, a can of worms perhaps.

On the other hand, changes to the UI is a never ending story and happening every day in the industry. UIs live, they are not meant to be consumed programmatically. It is not a matter of ToUs, it is a technical necessity for vitality.

BTW: Is this an issue or should it be converted to a discussion?

**Reply by fm-sys** [reply]
That would require a very, very big bandwidth for our server. And GC could easily block this IP.

And BTW, currently our users are violating the AGB's in theory when using c:geo. But GC cannot simply block more than one million geocacher (many of them premium members). But this would change completely, as soon as we interact with their website instead of our end users. In worst case, this would end up in expensive legal proceedings.

**Reply by fm-sys** [reply]
So a proxy isn't a way to go. Loading the regex dynamically from our server could work however. We could maybe even simply use github pages for it...?

**Reply by murggel** [reply]
I was not thinking of a proxy which takes over the complete scrapping but only to provide the regexpr. This has not be done on each call to the website, probably only few times a day. 

I don't know github pages, if this can not be manipulated easily, probably that is sufficient for downloading the regExpr

**Comment by murggel** [comment]
Can be converted to a discussion and reopened if necessary.

**Reply by baiti** [reply]
> Can be converted to a discussion and reopened if necessary.

Done
